subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/normalization/KotlinCompileClasspathFingerprinter.kt

```diff
@@ -37,7 +37,7 @@ class KotlinCompileClasspathFingerprinter(
     stringInterner: StringInterner
 ) : AbstractFileCollectionFingerprinter(
     ClasspathFingerprintingStrategy.compileClasspath(
-        CachingResourceHasher(AbiExtractingClasspathResourceHasher(KotlinApiClassExtractor()), cacheService),
+        CachingResourceHasher(AbiExtractingClasspathResourceHasher.withoutFallback(KotlinApiClassExtractor()), cacheService),
```

Comment:
I am curious why Kotlin does not use the fallback here.

Reply:
I don't pretend to fully understand, but they have explicit tests to ensure that, for instance, it doesn't recompile scripts in certain scenarios (e.g. bas class file).  With the input fingerprinting which this change targets, we're just changing the nature of the fallback (i.e. fallback to hash just the class instead of hashing the entire zip).  With the kotlin script classpath fingerprinting, there was no existing fallback at all, the exception would just boil up and be handling prior to compilation.  If we add this class-level fallback to the Kotlin script classpath fingerprinting, this would now not error out anymore but would silently hash the inputs and continue.  I assume that this would possibly create a different, more obscure failure message during compilation.

It's entirely possible that the kotlin stuff could/should be made to use this fallback as well, but it seemed like a separate concern from what I was trying to fix here.  So I decided to leave that functionality unchanged for now and figured we could address it later.

Reply:
Okay, that works for me.